content/WebAuthn/WebAuthn_Developer_Guide/webauthn-readiness-checklist.adoc

```diff
@@ -0,0 +1,198 @@
+
+_Note: Some sections might not apply to your solution._
+
+== Design
+
+image::ext-int-auth-webauthn-rp.png[]
+
+
+=== User Authentication Flows
+
+Select the user authentication flows your solution will support:
+
+* Single Factor Passwordless - Username and FIDO2 factor
+* Second Factor - Username, password, and FIDO second factor (U2F)
+* Multi-factor Authentication - FIDO2 factor and PIN
+
+
+=== Platforms
+
+Select the browsers and/or platforms your solution will support:
+
+==== Browsers (not an exhaustive list)
+
+* Android Browser
+* Brave
+* Brave on iOS
+* Chrome
+* Chrome for Android
+* Edge
+* Firefox
+* Firefox for Android
+* Opera
+* Safari
+
+==== Platforms
+
+* Windows
+* Mac
+* Linux
+* iOS
+* Android
+
+
+=== Authenticators
+
+Select the authenticator types, protocols, and transports your solution will support:
+
+
+==== Authenticator Types
+
+* Platform (Android biometric, Mac biometric, Windows Hello, etc…)
+* Cross-Platform (YubiKey, etc…)
+
+
+==== Authentication Protocols
+
+* CTAP1 (U2F)
+* CTAP2 (FIDO2)
+
+
+==== Transports
+
+* Bluetooth
+* Lightning
+* NFC
+* USB
+
+
+=== Extensions
+
+Both the WebAuthn and FIDO CTAP2 specifications provide for supporting a variety of extensions. Select the protocol extensions your solutions will support:
+
+
+==== WebAuthn Extensions:
+
+* FIDO AppID: use if migrating from U2F to WebAuthn.
+
+
+==== CTAP Extensions:
+
+* HMAC Secret
+* Cred Protect
+
+
+== Server-Side
+
+* This section applies only if your solution implements the WebAuthn relying party operations.
+
+
+=== User Verification
+
+To avoid unintended or unexpected user interaction, we recommend explicitly setting user verification (UV) for your specific solution’s use case.
+
+Because enabling user verification will cause the user to be prompted for a PIN to complete the WebAuthn operation, UV is helpful only for FIDO2 and WebAuthn implementations. For second factor flows, we recommend setting UV to discourage a PIN prompt.
+
+
+==== Scenario 1: WebAuthn and FIDO2 Implementations
+
+Request authenticator to require UV on registration and authentication
+
+* Set userVerification to “required” in the PublicKeyCredentialCreationOptions
+* Set userVerification to “required” in the PublicKeyCredentialRequestOptions
+
+
+==== Scenario 2: Second Factor Flows
+
+Request authenticator to minimize disruption in user interaction flow
+
+* Set userVerification to “discouraged” in the PublicKeyCredentialCreationOptions.
+
+  _Note: if the PIN has already been set, the user might be prompted to enter a PIN_
+
+* Set userVerification to “discouraged” in the PublicKeyCredentialRequestOptions
+
+
+=== Attestation
+
+The Attestation Object is applicable only to CTAP2 and WebAuthn implementations. Refer to the https://www.w3.org/TR/webauthn/#sctn-attestation[W3C standard] to learn more. We recommend always saving the raw attestation statement after every credential creation.
```

Comment:
U2F (CTAP1) indeed doesn't have the same attestation object, but it does provide an attestation certificate that's equally relevant to store.

Reply:
Could you please add in this information?

Reply:
Sorry, never mind! I forgot that the CTAP1 registration response is also packaged into the WebAuthn attestation object, so no change is needed here.